src/app/home-redesign/data.ts

```diff
@@ -23,13 +25,29 @@ export type UIStxBlock = Pick<
 export interface RecentBlocks {
   btcBlocks: GenericResponseType<UIBtcBlock[]>;
   stxBlocks: GenericResponseType<UIStxBlock[]>;
+  stxBlocksCountPerBtcBlock: Array<{
+    burn_block_time: string;
+    stx_blocks_count: number;
+    total_tx_count: number;
+  }>;
+}
+
+export interface UIStackingCycle {
+  cycleId: number;
+  stackedStx: number;
+  progressPercentage: number;
+  blocksTilNextCycle: number;
+  approximateDaysTilNextCycle: number;
+  approximateStartTimestamp: number;
+  approximateEndTimestamp: number;
+  rewardCycleLength: number;
+  startBlockHeight: number;
+  endBlockHeight: number;
 }
 
 export function fetchRecentBtcBlocks(chain: string, api?: string) {
   const apiUrl = api || getApiUrl(chain);
```

Comment:
this keeps getting repeated. Why not just get the api url outside this function and pass it in as the sole param?